unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+
+#pragma warning disable CS0162
+
+namespace DCL.Helpers.NFT.Markets.OpenSea_Internal
+{
+    internal class RequestController
+    {
+        internal const bool VERBOSE = false;
+
+        List<RequestGroup> requestGroup = new List<RequestGroup>();
+
+        public Request AddRequest(string assetContractAddress, string tokenId)
+        {
+            RequestGroup group = null;
+            for (int i = 0; i < requestGroup.Count; i++)
+            {
+                if (requestGroup[i].isOpen)
+                {
+                    group = requestGroup[i];
+                    break;
+                }
+            }
+            if (group == null)
+            {
+                group = CreateNewGroup();
+            }
+            return group.AddRequest(assetContractAddress, tokenId);
```

Comment:
Could it happen that a single assetContractAddress, tokenId pair is added more than once to the same request? Should we avoid this situation?

Reply:
> Could it happen that a single assetContractAddress, tokenId pair is added more than once to the same request? Should we avoid this situation?

It shouldn't happen, but It could happen.
In case it happen it won't break anything anyway :P